install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -220,10 +220,15 @@ spec:
           postStart:
             exec:
               command:
-              - "/cni-install.sh"
-              - "--enable-debug={{ .Values.debug.enabled }}"
-              - "--cni-exclusive={{ .Values.cni.exclusive }}"
-              - "--log-file={{ .Values.cni.logFile }}"
+              - "bash"
+              - "-c"
+              - |
+                /cni-install.sh --enable-debug={{ .Values.debug.enabled }}\
+                --cni-exclusive={{ .Values.cni.exclusive }}\
+                --log-file={{ .Values.cni.logFile }}
+                {{- if .Values.eni.enabled }}
+                {{- tpl (.Files.Get "files/agent/poststart-eni.bash") . | nindent 20 }}
```

Comment:
imo, we could move the script into the agent image, same as `/cni-install.sh`. The templating trick was probably only needed because we changed the mounting namespace and thus couldn't access the container fs anymore.

Reply:
That is a very fair point. I'm fine with moving the file in separate PR, so we can backport this one to stable branches and then require the newer agent images only in v1.14+